plugin/trino-hive/src/main/java/io/trino/plugin/hive/coercions/TimestampCoercer.java

```diff
@@ -97,4 +101,52 @@ protected void applyCoercedValue(BlockBuilder blockBuilder, Block block, int pos
                             toType));
         }
     }
+
+    public static class VarcharToShortTimestampCoercer
+            extends TypeCoercer<VarcharType, TimestampType>
+    {
+        public VarcharToShortTimestampCoercer(VarcharType fromType, TimestampType toType)
+        {
+            super(fromType, toType);
+        }
+
+        @Override
+        protected void applyCoercedValue(BlockBuilder blockBuilder, Block block, int position)
+        {
+            try {
+                Slice value = fromType.getSlice(block, position);
+                LocalDateTime dateTime = LocalDateTime.from(LOCAL_DATE_TIME.parse(value.toStringUtf8()));
+                toType.writeLong(
+                        blockBuilder,
+                        (dateTime.toEpochSecond(UTC) * MICROSECONDS_PER_SECOND) + floorDiv(dateTime.getNano(), NANOSECONDS_PER_MICROSECOND));
```

Comment:
- extract as variable (epochMicros)
- remove redundant braces
- `floorDiv` is redundant, since `getNano()` is non-negative

Reply:
> floorDiv

should this truncate or round? do you have test coverage showing hive behavior?

Reply:
I think truncation is fine - TestTimestampCoercer captures some error prone timestamp value.

Reply:
this is awesome!
note however that this is a unit test. and we want to ensure we're compatible with Hive behavior

same for returning null on failure -- we want to make sure Hive does that & does that the same way

Reply:
We have a value `2121` in product test where hive and trino returns null instead of throwing exception.

Reply:
that covers "null on failure", i guess.
what about truncate vs round?

Reply:
We could try to create a String with Nanoseconds precision which would allow us to verify if it performs truncate or round.

Reply:
Since `getNano` is positive I'm not sure if a value like this `2121-07-15 15:30:12.999999999` we give us a situation where we would round.

Reply:
In case of hive - it doesn't have a parameterized timestamp type - so there isn't any rounding off between a timestamp(3) or timestamp(6) or timestamp(9). So either we need to restrict the coercion only for nano-seconds precision or we need to mimic the behavior similar to reading a timestamp column - which was written in a nanoseconds. WDYT ?

Reply:
> or we need to mimic the behavior similar to reading a timestamp column - which was written in a nanoseconds.

agreed, we should do that